src/dialog/dlgkeywheel.cpp

```diff
@@ -0,0 +1,153 @@
+#include "dialog/dlgkeywheel.h"
+
+#include <QDir>
+#include <QKeyEvent>
+#include <QMouseEvent>
+
+#include "control/controlobject.h"
+
+using namespace mixxx::track::io::key;
+
+namespace {
+auto kKeywheelSVG = QStringLiteral("images/keywheel/keywheel.svg");
+const KeyUtils::KeyNotation kNotationHidden[]{
+        KeyUtils::KeyNotation::OpenKeyAndTraditional,
+        KeyUtils::KeyNotation::LancelotAndTraditional};
+} // namespace
+
+DlgKeywheel::DlgKeywheel(QWidget* parent, UserSettingsPointer pConfig)
+        : QDialog(parent),
+          ui(new Ui::DlgKeywheel),
+          m_pConfig(pConfig) {
+    ui->setupUi(this);
+    QDir resourceDir(m_pConfig->getResourcePath());
+    auto svgPath = resourceDir.filePath(kKeywheelSVG);
+    qDebug() << svgPath;
+    QFile xmlFile(svgPath);
+    if (!xmlFile.exists() || !xmlFile.open(QFile::ReadOnly | QFile::Text)) {
+        qDebug() << "Could not load template: " << svgPath;
+        return;
+    }
+    m_domDocument.setContent(&xmlFile);
+
+    // FIXME(poelzi): try to prevent aspect ratio. This fails unfortunatelly
+    QSizePolicy qsp = ui->graphic->sizePolicy(); //.setHeightForWidth(true);
+    qsp.setHeightForWidth(true);
+    ui->graphic->setSizePolicy(qsp);
+    installEventFilter(this);
+    ui->graphic->installEventFilter(this);
+
+    // load the user configured setting as default
+    const int notation = static_cast<int>(ControlObject::get(
+            ConfigKey("[Library]", "key_notation")));
+    m_notation = static_cast<KeyUtils::KeyNotation>(notation);
+    // Select a valid display and update
+    switchDisplay(0);
+}
+
+bool DlgKeywheel::eventFilter(QObject* obj, QEvent* event) {
+    if (event->type() == QEvent::KeyPress) {
+        // we handle TAB + Shift TAB to cycle through the display types
+        QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
+        if (keyEvent->key() == Qt::Key_Tab) {
+            switchDisplay(+1);
+            return true;
+        } else if (keyEvent->key() == Qt::Key_Backtab) {
+            switchDisplay(-1);
+            return true;
+        }
+    } else if (event->type() == QEvent::MouseButtonPress) {
+        QMouseEvent* mouseEvent = static_cast<QMouseEvent*>(event);
+        // first button forward, other buttons backward cycle
+        switchDisplay(mouseEvent->button() == Qt::LeftButton ? +1 : -1);
+        return true;
+    }
+    // standard event processing
+    return QObject::eventFilter(obj, event);
```

Comment:
I think this should be
```suggestion
    return QDialog::eventFilter(obj, event);
```
..,to not jump the inheritance.